mlflow/pmdarima.py

```diff
@@ -0,0 +1,364 @@
+    """
+    Load a ``pmdarima`` ``ARIMA`` model or ``Pipeline`` object from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                      - ``mlflow-artifacts:/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+                      artifact-locations>`_.
+    :param dst_path: The local filesystem path to which to download the model artifact.
+                     This directory must already exist. If unspecified, a local output
+                     path will be created.
+
+    :return: A ``pmdarima`` model instance
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
+    pmdarima_model_file_path = os.path.join(
+        local_model_path, flavor_conf.get(_MODEL_BINARY_KEY, _MODEL_BINARY_FILE_NAME)
+    )
+
+    return _load_model(pmdarima_model_file_path)
+
+
+def _save_model(model, path):
+
+    with open(path, "wb") as f:
+        pickle.dump(model, f)
+
+
+def _load_model(path):
+
+    with open(path, "rb") as pickled_model:
+        model = pickle.load(pickled_model)
+    return model
+
+
+def _load_pyfunc(path):
+
+    return _PmdarimaModelWrapper(_load_model(path))
+
+
+class _PmdarimaModelWrapper:
+    def __init__(self, pmdarima_model):
+        import pmdarima
+
+        self.pmdarima_model = pmdarima_model
+        self._pmdarima_version = pmdarima.__version__
+
+    def predict(self, dataframe) -> pd.DataFrame:
+
+        schema = {"n_periods", "X", "return_conf_int", "alpha"}
+        df_schema = dataframe.columns.values.tolist()
+
+        if len(dataframe) > 1:
+            raise MlflowException(
+                f"The provided prediction pd.DataFrame {dataframe.to_string()} "
+                f"contains {len(dataframe)} rows. Only 1 row should be supplied."
+            )
+
+        if not set(df_schema).issubset(schema):
+            raise MlflowException(
+                f"The provided schema {df_schema} contains invalid columns. "
+                f"Columns must be part of: {schema}"
+            )
+
+        attrs = dataframe.to_dict(orient="index").get(0)
+        n_periods = attrs.get("n_periods", None)
+
+        # NB Any model that is trained with exogeneous regressor elements will need to provide
+        # `X` entries as a 2D array structure to the predict method.
+        exogoneous_regressor = attrs.get("X", None)
+
+        if exogoneous_regressor and self._pmdarima_version < "1.8.0":
```

Comment:
```suggestion
        if exogoneous_regressor and Version(self._pmdarima_version) < Version("1.8.0"):
```

Can we use `packaging.version.Version` here?

```python
>>> from packaging.version import Version
>>> "1.8.0.rc1" > "1.8.0"
True
>>> Version("1.8.0.rc1") > Version("1.8.0")
False
```